flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java

```diff
@@ -228,10 +227,8 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
                                         flushMode,
                                         objectReuse
                                                 ? createRowDataTypeSerializer(
-                                                                context,
-                                                                dataStream.getExecutionConfig())
-                                                        ::copy
-                                                : Function.identity());
```

Comment:
Just curious why we need to remove the following

```
:copy
: Function.identity());
```

Reply:
None of those functions is serializable therefore we cannot pass them directly to the sink.

Reply:
While `Function.identity()` is not `Serializable`, `x -> x` is when created for a `SerializableFunction`.